buildtesters/buildtest
# json schema for buildtest configuration

This PR makes first attempt to do the following:
- Provides a json schema for buildtest configuration (``config_schema.json``)
- buildtest configuration is now changed to json file (``settings.json``)
- Add regtest for schema check
- Remove ``BUILDTEST_VERSION`` from config_opts, it is not needed
- Update ``config_opts["EDITOR"]`` to ``config_opts["editor"]``  that relates to ``buildtest config edit``
- Remove option ``buildtest show --config`` see https://github.com/HPC-buildtest/buildtest-framework/pull/236/commits/8b0f4f303c4a40e7d60c2bcfbc8bd46f77e328ec
- Initial support for validating scheduler and launcher types. If launcher is not found in $PATH, we exit immediately.
